File: print-numbers-from-n-to-1-without-using-a-loop/PrintNumbersFromNTo1WithoutUsingALoop.cpp
// Write a program, that prints numbers from n to 1 without using a loop
#include <iostream>

// Instead of using a loop, this function just prints the current number and
// calls itself again with the number - 1(Recursion).
void PrintNum(int);

int main()
{
    PrintNum(5);
    return 0;
}

void PrintNum(int n)
{
    // base case
    if (n > 0)
    {
        std::cout << n << " ";
        PrintNum(n - 1);
    }
    else
        std::cout << std::endl;

    return;
}

File: program/program/print-numbers-from-n-to-1-without-using-a-loop/print_numbers_from_n_to_1_without_using_a_loop.go
package main

import (
	"fmt"
)

func main() {
	fmt.Printf("Enter a Number:")
	var n int
	fmt.Scanf("%d", &n)
	ans(n)
	fmt.Print("\n")
}
func ans(n int) {
	//without loop
	//using recursion
	if n < 1 {
		return
	}
	fmt.Printf("%d ", n)
	ans(n - 1)
}
